packages/cli/src/commands/screenshot-diff/screenshot-diff.command.ts

```diff
@@ -18,199 +22,212 @@ import {
-  ) {
-    super(message);
-  }
-}
-
-type ComparisonOutcome = "pass" | "fail";
-
-export interface ScreenshotDiffResult {
-  baseScreenshotFile: string;
-  headScreenshotFile: string;
-
-  outcome: ComparisonOutcome;
-  comparisonResult: CompareImageResult;
-}
-
-export const diffScreenshots: (options: {
-  client: AxiosInstance;
+export const diffScreenshots = async ({
+  headReplayId,
+  baseReplayId,
+  headScreenshotsDir,
+  baseScreenshotsDir,
+  diffOptions,
+}: {
   baseReplayId: string;
   headReplayId: string;
   baseScreenshotsDir: string;
   headScreenshotsDir: string;
   diffOptions: ScreenshotDiffOptions;
-  exitOnMismatch: boolean;
-}) => Promise<ScreenshotDiffResult[]> = async ({
-  client,
-  baseReplayId,
-  headReplayId,
-  baseScreenshotsDir,
-  headScreenshotsDir,
-  diffOptions,
-  exitOnMismatch,
-}) => {
+}): Promise<ScreenshotDiffResult[]> => {
   const { diffThreshold, diffPixelThreshold } = diffOptions;
-  const logger = log.getLogger(METICULOUS_LOGGER_NAME);
 
   const baseReplayScreenshots = await getScreenshotFiles(baseScreenshotsDir);
   const headReplayScreenshots = await getScreenshotFiles(headScreenshotsDir);
 
-  // Assume base replay screenshots are always a subset of the head replay screenshots.
-  // We report any missing base replay screenshots for visibility but don't count it as a difference.
   const missingHeadImages = new Set(
-    [...baseReplayScreenshots].filter(
+    baseReplayScreenshots.filter(
       (file) => !headReplayScreenshots.includes(file)
     )
   );
 
-  let totalMismatchPixels = 0;
-
-  const comparisonResults: ScreenshotDiffResult[] = [];
-
-  try {
-    if (missingHeadImages.size > 0) {
-      logComparisonResultMessage(
-        logger,
-        `Head replay is missing screenshots: ${[...missingHeadImages].sort()}`,
-        "fail"
-      );
-      throw new DiffError(
-        `Head replay is missing screenshots: ${[...missingHeadImages].sort()}`,
-        {
-          baseReplayId,
-          headReplayId,
-          threshold: diffThreshold,
-        }
-      );
+  const missingHeadImagesResults: ScreenshotDiffResult[] = Array.from(
+    missingHeadImages
+  ).flatMap((screenshotFileName) => {
+    const identifier = getScreenshotIdentifier(screenshotFileName);
+    if (identifier == null) {
+      return [];
     }
+    return [
+      {
+        identifier,
+        outcome: "missing-head",
+        baseScreenshotFile: `screenshots/${screenshotFileName}`,
+      },
+    ];
+  });
 
-    for (const screenshotFileName of headReplayScreenshots) {
-      if (!baseReplayScreenshots.includes(screenshotFileName)) {
-        logger.info(
-          `Screenshot ${screenshotFileName} not present in base replay`
-        );
-        continue;
-      }
+  const headDiffResults = (
+    await Promise.all(
+      headReplayScreenshots.map(
+        async (screenshotFileName): Promise<ScreenshotDiffResult[]> => {
+          const identifier = getScreenshotIdentifier(screenshotFileName);
```

Comment:
[nit / not important] Could also pull this out as a function below if you want to reduce nesting.

Reply:
I would have to pass around 6 parameters or so, so I will leave it like this.

Reply:
Ack, that makes sense. The other trick I sometimes do in that kind of case if I want to reduce nesting is to pull it out as a function but within the other function:

```ts
  const diffWithCorrespondingBaseScreenshot = (headScreenshotFileName: string): Promise<ScreenshotDiffResult[]> => {

      // ... long function body

  };

  const headDiffResults = await Promise.all(headReplayScreenshots.map(diffWithCorrespondingBaseScreenshot).flat()));
```

That way it's the function has a name, so it's easier to understand what it's doing (you don't have to read the whole function body to get the gist of what it's doing in order to understand the rest of the code); the .flat() becomes more visible, rather than hidden at the bottom; and there's less tabbing, which often makes it a little easier to follow.

(other way of course is to use an options interface, but that can be a pain)

Reply:
Done in c8ca30dd7105121e118b5a532208a8c8f75fc947